### ISSUE TITLE
update dev or rc branches per CFEP-05

### ISSUE THREAD

**Issue by beckermr** [regro/cf-scripts#763]
We should support version updates on `dev` or `rc` branches per [CFEP-05](https://github.com/conda-forge/cfep/pull/3). 

IDK exactly what semantics we'd like to follow wrt finding updated versions, but it seems like we could at least catch a few.

**Comment by CJ-Wright** [comment]
Yeah, I think we might want an opt in system did this or the maintainers might get a ton of PRs. Also we need to point the PRs at a different branch and have them not count against the normal PR limits and version bump criteria.

**Comment by CJ-Wright** [comment]
I think a good first use case for this would be python
